src/main/scala/mesosphere/marathon/api/akkahttp/v2/TasksController.scala

```diff
@@ -100,14 +101,23 @@ class TasksController(
         }
       }
   }
-  private def getTasksByAppId(instanceIdsToAppId: Map[Id, PathId])(implicit identity: Identity): Future[Map[PathId, Seq[Instance]]] = {
+
+  private def getTasksToKill(instanceIdsToAppId: Map[Id, PathId], affectedApps: IndexedSeq[AppDefinition])(implicit identity: Identity): Future[Map[PathId, Seq[Instance]]] = async {
+    /**
+      * Pods can't be killed using /tasks endpoint, only apps
+      */
+    def isNotPod(appId: PathId): Boolean = affectedApps.exists(app => app.id == appId)
```

Comment:
How is `affectedApps.exists(app => app.id == appId)` verifying that the app is not a pod?

Reply:
from what I understood from @unterstein affected apps is collection of apps that was retrieved via groupManager.app which should return None if the PathId actually belong to Pod...

Any idea how to have this more obvious ... ?